settings.lua

```diff
@@ -2834,6 +2825,27 @@ function RealisticDrivingSetting:init(vehicle)
 	self:set(true)
 end
 
+---@class DriveUnloadNowSetting : BooleanSetting
+DriveUnloadNowSetting = CpObject(BooleanSetting)
+function DriveUnloadNowSetting:init(vehicle)
+	BooleanSetting.init(self, 'driveUnloadNow', 'COURSEPLAY_DRIVE_NOW', 'COURSEPLAY_DRIVE_NOW', vehicle)
+	self:set(false)
+end
+
+function DriveUnloadNowSetting:onChange()
+	courseplay.hud:setReloadPageOrder(self.vehicle, self.vehicle.cp.hud.currentPage, true)
```

Comment:
I'm not a big fan of this, creates a mutual dependency, a Setting should not know anything about the HUD.

Reply:
There seems to be a few problems that the hud doesn't refresh, do you have any idea on how to correct them better ?

Reply:
What exact scenario is that when it does not get refreshed?

Reply:
While being in an CombineAIDriver and he finds a combine the page for Unloaders sometimes doesn't get refreshed or once I've called a driver .The Drive Now button sometimes doesn't get refreshed, the last one might be more of a MP problem.

Reply:
should fix these: #4820 and #5558

Reply:
If we talk about HUD Issues, let me make a list of things that is not working correctly even in SP. There is more 'refresh' stuff and strange bugs about it, before we talk about MP related bugs...

Reply:
@pvaiko  one last question for HUD refresh, what do you would be a good point to decide if the hud should be reloaded ? 
@Tensuko yeah you're right just found these bugs while testing and search for a solution on the way

Reply:
@schwiti6190 I really don't want to have these in the Setting, which is the model in the classic MVC pattern, the HUD is the view, it would really be the job of the controller to refresh the view (I know, CP does not really follow any patterns). 

Still, I think it would be better to resolve these problems by initiating the HUD refresh from the place where the Setting was changed.

Reply:
I understand your idea and position for this problem, but now it seems like a lot of code duplication. Sadly I've haven't found a good solution myself.